public/app/plugins/panel/timeline/timeline.ts

```diff
@@ -97,6 +97,8 @@ export function getConfig(opts: TimelineCoreOptions) {
   const font = `500 ${Math.round(12 * devicePixelRatio)}px ${theme.typography.fontFamily}`;
   const hovered: Array<Rect | null> = Array(numSeries).fill(null);
 
+  const hzGap = 0;
```

Comment:
future option? was this the thing you wanted to build into uPlot or the distribute function?

Reply:
in periodic mode gaps are okay, but in state changes mode they affect accuracy. even a 2px gap looks like a rendering mistake where state edges don't land exactly on gridlines. this is especially noticeable when consecutive states are not densly packed.

the distribute function in timeline is only used in the vertical direction to shift the series apart into rows, so this is not quire the same, but i left this here to figure out what to do in a follow-up, post panel split. by default i dont think there should be gaps in state change mode, since edge accuracy is important. i found myself having to turn it off when working on this PR to make sure it wasnt an actual rendering bug.

Reply:
> state edges don't land exactly on gridlines

but if the gap is on the right side that should be ok? a 1-2 px gap there should be ok I think, but not sure, the gaps really helped make the transitions more clear

Reply:
I really think we need to split the two modes into two panels -- the distinction when this makes sense really matters